frontend/src/compositions/CartProduct/CartProduct.tsx

```diff
@@ -0,0 +1,53 @@
+import { Minus, Plus, Trash } from "lucide-react";
+
+import { Button, Input, Label, Text } from "@/components/retroui";
+
+export function CartProduct({ item }: { item: CartItem }) {
```

Comment:
💭 I think some properties could be de-structured out of `item` to avoid having so much chaining happening. I also think `item` could be more descriptive like `cartProductItem` or something.

```suggestion
const {quantity, product} = item
```

saves you from having to do `item.product.xxx` many times. 

I also think we need some optional chaining where you are diving into those object properties a few layers.
like `product?.price?.value` etc